src/main/java/org/terasology/fluid/system/FluidClientSystem.java

```diff
@@ -122,4 +130,45 @@ private void setFluidContainerIcon(EntityRef container, FluidContainerItemCompon
             }
         }
     }
+
+    /**
+     * Used to draw the Filling bar over the fluid container item in the cell.
+     *
+     * @param event  An event sent after the inventory cell has been rendered.
+     * @param entity The entity sending the request.
+     * @param fluidContainer FluidContainerItemComponent of the item.
+     */
+    @ReceiveEvent
+    public void drawFillingBarForFluidContainerItem(InventoryCellRendered event, EntityRef entity,
+                                                    FluidContainerItemComponent fluidContainer) {
+        Canvas canvas = event.getCanvas();
+
+        Vector2i size = canvas.size();
+
+        int minX = (int) (size.x * 0.8f);
+        int maxX = (int) (size.x * 0.9f);
+
+        int minY = (int) (size.y * 0.1f);
+        int maxY = (int) (size.y * 0.9f);
+
+        float fillingPercentage = 1f * fluidContainer.volume / fluidContainer.maxVolume;
+
+        if (fillingPercentage != 1f) {
+            ResourceUrn backgroundTexture = TextureUtil.getTextureUriForColor(Color.WHITE);
+            final Color terasologyColor = getTerasologyColorForFilledAmount(fillingPercentage);
```

Comment:
I'd prefer to use a simple blue color for the filling. If you want to stick to a gradient, maybe we can use one from dark to light blue (change the **B**rightness instead of the **H**ue in HSB color).

Reply:
The purpose of gradient was to have different colors(red, yellow, green) conveying the amount filled just by looking at the color... red signifying dangerously low or near empty .... yellow meaning -  you've got some left, green - you've got enough water or near full...
If we're using a blue color, I don't think the gradient would be much help.
So I'll go ahead with a simple blue color.